dartagnan/src/main/java/com/dat3m/dartagnan/utils/GitInfo.java

```diff
@@ -11,19 +11,31 @@
 
 public class GitInfo {
 
-	private static final Logger logger = LogManager.getLogger(Dartagnan.class);  
+    private static final Logger logger = LogManager.getLogger(Dartagnan.class);
 
-	static Properties properties = new Properties();
-	
-	public static void CreateGitInfo() throws IOException {
-       try (InputStream is = Dartagnan.class.getClassLoader()
+    static Properties properties = new Properties();
```

Comment:
Why do we have this static field if it gets effectively reassigned/overwritten in all operations?
Can't we load this once and then have the `CreateGitInfo/getGidId` (btw. the first method's name should be lower case) read out the data?

Alternatively, you can remove this field and provide a `private static Properties loadGitProperties()` method that loads the properties object (possibly `null` on failure?). Then we can later decide if this helper function should load the properties only once and cache the result.

Reply:
> Why do we have this static field if it gets effectively reassigned/overwritten in all operations? 

The method is static otherwise the compiler complains about `Cannot make a static reference to the non-static field properties`. Also, I don't think loading information (which is what `properties.load` does) counts as reassigning/overwritting

Reply:
No, I mean why don't you just have a local `Properties properties = new Properties()` in each function to avoid this "global" variable?

Reply:
If we move the variable to each method, we need to repeat the loading each time. Now the init() method takes care of the loading and the other methods just query the properties they need.

Reply:
Yeah, that's fine. Either do it once globally or each time locally; those were my two suggestions :).